utils/HelpAbstractionGenerator/StepBehaviors/Step4.WebObjectLibrary.cs

```diff
@@ -0,0 +1,1040 @@
+                vtype = new ejoValueType() { id = sig[0].Trim().ToLower(), typeNames = "Number", primitive = "real" };
+            else if (sig[1].Contains("INTEGER") == true || sig[1].Contains("COLORNAME") == true)
+                vtype = new ejoValueType() { id = sig[0].Trim().ToLower(), typeNames = "Number", primitive = "integer" };
+            else if (sig[1].Contains("STRING") == true || sig[1].Contains("CHAR") == true)
+                vtype = new ejoValueType() { id = sig[0].Trim().ToLower(), typeNames = "string", primitive = "string" };
+            else if (sig[1].Contains("FUNCTION") == true)
+                vtype = new ejoValueType() { id = sig[0].Trim().ToLower(), typeNames = "LispExpression", primitive = "string" };
+            else if (template.Title.ToUpper() == "LAYOUT")
+                vtype = new ejoValueType() { id = sig[0].Trim().ToLower(), typeNames = sig[1].Trim().Trim(';').ToLower(), primitive = "enum", enums = new string[] { "horizontal", "vertical" } };
+            else
+                vtype = new ejoValueType() { id = sig[0].Trim().ToLower(), typeNames = sig[1].Trim().Trim(';').ToLower(), primitive = "enum", enums = new string[] { "left", "right", "top", "bottom", "centered" } };
+            var result = new ejoAttribute()
+            {
+                category = ejoCategory.DCLATT,
+                description = template.Description,
+                guid = Source.guid,
+                id = template.getProfile(ejoCategory.DCLATT, false),
+                platforms = template.Platforms,
+                signature = template.DclSignature,
+                valueType = vtype
+            };
+            jsonString = result.toJSON(Newtonsoft.Json.Formatting.Indented);
+            return result;
+        }
+
+        
+        public List<ejoFunction> parsePropertyPage()
+        {
+            HAP.HtmlNode sigNode = template.signatureHeaders.First();
+            string types = sigNode.getDoc().SelectNodes(sigNode.ParentNode.XPath + "//p").ToList().Find(p => p.InnerText.getLetters().StartsWith("TYPE")).InnerText.Normalization();
+            var obj = new ejoValueType("Object", types, null, linkCategory.Object);
+
+            HAP.HtmlNode retNode = template.returnValHeaders.First();
+            HAP.HtmlNode ulNode = retNode.getDoc().SelectSingleNode(retNode.ParentNode.XPath + "//ul");
+            string readonlyValue = retNode.nextRealSibling().InnerText.getLetters();
+            string typeValue = string.Join(", ", retNode.nextRealSibling().nextRealSibling().InnerText.Normalization().splitByAnyChar(" ,"));
+
+            var outVal = new ejoValueType("Return", typeValue, ulNode, linkCategory.Property);
+            var inVal = new ejoValueType("NewValue", typeValue, ulNode, linkCategory.Property);
+
+            bool isReadOnly = false;
+            if (readonlyValue.Contains("READONLYYES") && readonlyValue.Contains("EXCEPT") == false)
```

Comment:
Probably missing some vla-set functions that are valid simply because of the EXCEPT rule. Depending on how this gets used in the future, I probably need to create the setters, but somehow find the things they are excluding and also include them in our setters.